lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -679,6 +680,59 @@ def create_common_methods(common_relation_methods_module)
             end
           end
 
+          BATCHES_METHODS.each do |method_name|
+            case method_name
+            when :find_each
+              create_common_method(
+                "find_each",
+                common_relation_methods_module,
+                parameters: [
+                  create_kw_opt_param("start", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("end", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("batch_size", type: "Integer", default: "1000"),
+                  create_kw_opt_param("error_on_ignore", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("order", type: "Symbol", default: ":asc"),
+                  create_block_param("block", type: "T.nilable(T.proc.params(object: #{constant_name}).void)"),
+                ],
+                return_type: "T.nilable(T::Enumerable[#{constant_name}])",
+              )
+            when :find_in_batches
+              create_common_method(
+                "find_in_batches",
+                common_relation_methods_module,
+                parameters: [
+                  create_kw_opt_param("start", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("end", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("batch_size", type: "Integer", default: "1000"),
+                  create_kw_opt_param("error_on_ignore", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("order", type: "Symbol", default: ":asc"),
+                  create_block_param(
+                    "block",
+                    type: "T.nilable(T.proc.params(object: T::Array[#{constant_name}]).void)",
+                  ),
+                ],
+                return_type: "T.nilable(T::Enumerable[T::Enumerable[#{constant_name}]])",
+              )
+            when :in_batches
+              use_ranges = ActiveRecord::Batches.instance_method(:find_each).parameters.include?([:key, :use_ranges])
```

Comment:
Not related to this PR but it may be worth investing in a `rails_helper` similar to https://github.com/Shopify/tapioca/blob/main/lib/tapioca/helpers/sorbet_helper.rb if these keep coming up. We had a similar occurrence in https://github.com/Shopify/tapioca/pull/1327.